### PR TITLE
Remove System.Linq dependency from System.Security.Cryptography.Encoding

### PR DESCRIPTION
**Pull request by stephentoub**

### CODE REVIEW

File: src/System.Security.Cryptography.Encoding/src/Internal/Cryptography/OidLookup.OSX.cs [hunk 1]
```diff
@@ -53,7 +52,7 @@ private static string NativeFriendlyNameToOid(string friendlyName, OidGroup oidG
         // and OpenSSL produce different answers.  Since the answers originally came from OpenSSL
         // on macOS, this preserves the OpenSSL names.
         private static readonly Dictionary<string, string> s_extraFriendlyNameToOid =
-            new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase)
+            new Dictionary<string, string>(8, StringComparer.OrdinalIgnoreCase)
```
Comment (davidsh) [inline src/System.Security.Cryptography.Encoding/src/Internal/Cryptography/OidLookup.OSX.cs]: where does "8" come from?

File: src/System.Security.Cryptography.Encoding/src/Internal/Cryptography/OidLookup.cs [hunk 2]
```diff
@@ -108,7 +107,7 @@ public static string ToOid(string friendlyName, OidGroup oidGroup, bool fallBack
         // is to prevent issues wherein an identifier is different between CoreFX\Windows and CoreFX\Unix;
         // since any existing code would be using the Windows identifier, it is the de facto standard.
         private static readonly Dictionary<string, string> s_friendlyNameToOid =
-            new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase)
+            new Dictionary<string, string>(103, StringComparer.OrdinalIgnoreCase)
```
Comment (davidsh) [inline src/System.Security.Cryptography.Encoding/src/Internal/Cryptography/OidLookup.cs]: where does "103" come from?

File: src/System.Security.Cryptography.Encoding/src/Internal/Cryptography/OidLookup.OSX.cs (hunk 1 above)
Comment (stephentoub) [reply src/System.Security.Cryptography.Encoding/src/Internal/Cryptography/OidLookup.OSX.cs]: The number of items being added immediately after this line.  This is just pre-sizing it.

File: src/System.Security.Cryptography.Encoding/src/Internal/Cryptography/OidLookup.cs (hunk 2 above)
Comment (stephentoub) [reply src/System.Security.Cryptography.Encoding/src/Internal/Cryptography/OidLookup.cs]: The number of items being added immediately after this line.  This is just pre-sizing it.

File: src/System.Security.Cryptography.Encoding/src/Internal/Cryptography/OidLookup.OSX.cs (hunk 1 above)
Comment (bartonjs) [inline src/System.Security.Cryptography.Encoding/src/Internal/Cryptography/OidLookup.OSX.cs]: Consider making the 8 a constant and adding a #if DEBUG cctor that asserts that the count is correct (both here and the other one)

Comment (stephentoub) [reply src/System.Security.Cryptography.Encoding/src/Internal/Cryptography/OidLookup.OSX.cs]: Ok

**Pull request merged**